OpenSim/Framework/PrimitiveBaseShape.cs

```diff
@@ -1051,6 +1071,13 @@ public ulong GetMeshKey(Vector3 size, float lod)
                 hash = djb2(hash, this.SculptType);
             }
 
+            if (this.RenderMaterials != null)
```

Comment:
I dont think anything about render materials affect the physics shape. If this is the case, the materials should not be included in the hash since two objects with different materials would still have the same shape.